src/charm.py

```diff
@@ -140,349 +154,143 @@ def __init__(self, *args):
 
         self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
         self.framework.observe(self.ingress.on.revoked, self._on_ingress_revoked)
-
+        self.framework.observe(
+            self.openfga_requirer.on.openfga_store_created,
+            self._on_openfga_store_created,
+        )
```

Comment:
i would have expected the `Integration` objects to "swallow" the observe part by defining the hooks in a `RegisterEvent` way

that would separate logic into each own integration object

what i dunno though is if `observe` allows multiple hooks on the same event 

if we cant do the following then feel free to trash my point
```
self.framework.observe(self.ingress.on.revoked, self._fx)
self.framework.observe(self.ingress.on.revoked, self.__fy)
```

Reply:
I would assume this is not possible for registering multiple callbacks to a single event (but correct me if I'm wrong).

I think it's good to leave those hook registering in the `charm.py` since the `charm.py` is mainly responsible for dealing with the hook callbacks. Code readers can find the event -> callback mappings directly here.

Reply:
It is possible to register multiple callbacks for a single event. Some libraries do this, where the listen for example for pebble ready. In the past the documentation around how it works was not there (for example what happens if you defer in one of the handlers? do all of them get deferred?).